src/storage/resourceManager.ts

```diff
@@ -695,17 +700,17 @@ export class ResourceManager {
     const key = SecretStorageKeys.DIRECT_CONNECTIONS;
     return await this.runWithMutex(key, async () => {
       const connectionIds: DirectConnectionsById = await this.getDirectConnections();
-      connectionIds.set(connection.id!, connection);
+      connectionIds.set(connection.id! as ConnectionId, connection);
       await this.storage.setSecret(key, JSON.stringify(Object.fromEntries(connectionIds)));
     });
   }
 
-  async deleteDirectConnection(id: string): Promise<void> {
+  async deleteDirectConnection(id: ConnectionId): Promise<void> {
     const key = SecretStorageKeys.DIRECT_CONNECTIONS;
     return await this.runWithMutex(key, async () => {
       const connections: DirectConnectionsById = await this.getDirectConnections();
       connections.delete(id);
-      await this.storage.setSecret(key, JSON.stringify(connections));
+      await this.storage.setSecret(key, JSON.stringify(Object.fromEntries(connections)));
```

Comment:
Bug fix from previous manager branch (but we weren't actually deleting anything back then) that would accidentally delete all direct connections instead of just the one, by the power of `JSON.stringify(<map>)` turning the map into `"{}"`

Reply:
👍